cms/djangoapps/contentstore/course_info_model.py

```diff
@@ -131,17 +137,15 @@ def delete_course_update(location, update, passed_id, user):
     passed_index = _get_index(passed_id)
 
     # delete update item from given index
-    if 0 < passed_index <= len(course_update_items):
-        course_update_item = course_update_items[passed_index - 1]
-        # soft delete course update item
-        course_update_item["status"] = CourseInfoBlock.STATUS_DELETED
-        course_update_items[passed_index - 1] = course_update_item
-
-        # update db record
-        save_course_update_items(location, course_updates, course_update_items, user)
-        return _get_visible_update(course_update_items)
-    else:
-        return HttpResponseBadRequest(_("Invalid course update id."))
+    for course_update_item in course_update_items:
+        if course_update_item["id"] == passed_index:
```

Comment:
How will this condition cater for `passed_index - 1` being done previously?

Reply:
Previously, this condition relies on the structure of the data and it was causing troubles with some data e.g. a single course update with an id of 2. This `passed_index - 1` will result in 1 which is an invalid id.

Reply:
I am still not clear on the explanation, but given the tests are passing, I am going to give it a thumbs up.